### PR TITLE
Pytorch triton merge

### PR DESCRIPTION
**Pull request by dllehr-amd**

### CODE REVIEW

**Comment by jayfurmanek** [comment]
ok - we've merged in a number of new PRs. Can you fix the conflict here and we'll get this in as well.

**Review by jataylo — COMMENTED** [review]
LGTM but if our eventual goal is to merge into triton-mlir we need to bring in the github workflow scripts as this is cloned on upstream pytorch for the wheel building ci.

File: python/triton/common/build.py [hunk 1]
```diff
@@ -24,7 +24,13 @@ def libcuda_dirs():
 
 @functools.lru_cache()
 def rocm_path_dir():
-    return os.getenv("ROCM_PATH", default="/opt/rocm")
+    default_path = os.path.join(os.path.dirname(__file__), "third_party", "rocm")
```
[the remark this answers is not in the document]
Comment (jataylo) [reply python/triton/common/build.py]: This is now merged in

**Comment by jataylo** [comment]
Change came in via https://github.com/ROCmSoftwarePlatform/triton/pull/354 instead

**Pull request closed without merging**